core/src/apps/cardano/layout.py

```diff
@@ -347,23 +347,50 @@ async def show_device_owned_output_credentials(
     stake_credential: Credential,
     show_both_credentials: bool,
 ) -> None:
-    intro_text = "The following address is owned by this device. Its"
-    await _show_credential(ctx, payment_credential, intro_text, is_output=True)
+    intro_text = "The following address is owned by this device."
+    await _show_credential(ctx, payment_credential, intro_text, purpose="output")
     if show_both_credentials:
-        await _show_credential(ctx, stake_credential, intro_text, is_output=True)
+        await _show_credential(ctx, stake_credential, intro_text, purpose="output")
+
+
+async def show_governance_registration_reward_credentials(
+    ctx: Context,
+    payment_credential: Credential,
+    stake_credential: Credential,
+    show_both_credentials: bool,
+    show_payment_warning: bool,
+) -> None:
+    intro_text = "The governance registration reward address is owned by this device."
+    await _show_credential(
+        ctx, payment_credential, intro_text, purpose="gov_reg_reward_address"
+    )
+    if show_both_credentials or show_payment_warning:
+        extra_text = (
+            "Warning: The address is not a payment address, it is not eligible for rewards."
```

Comment:
Let's extract this into a constant since the same string is used below in `confirm_governance_registration_reward_address`.